### ISSUE TITLE
SIM card in 4G USB Dongle not recognized

### ISSUE THREAD

**Issue by filmmaker75** [Ysurac/openmptcprouter#3154]
## Expected Behavior
I have two perfectly working HUAWEI 3372-325 dongles, I would like to add the third.

## Current Behavior
I have currently successfully configured two HUAWEI E3372-325 USB dongles with two SIM cards from two different carriers. Everything runs on a Raspberry PI4 and a VPS on Aruba, it works perfectly and the aggregation happens correctly. I purchased a third HUAWEI E3372-325 4G USB dongle and a third SIM from another provider to increase the bandwidth. Unfortunately when I connect the latter to the Raspberry the LED flashes green/red and the interface is not recognised. If I try the dongle in the PC everything works fine. I also tried swapping SIMs and dongles but to no avail. It seems that the new SIM is not recognized by OpenMPTCProuter and consequently the relevant USB interface is not created. Can I solve it somehow? Thank you.

## Specifications

  - OpenMPTCProuter version: 0.59.1-5.4 r0+16594-ce92de8c8c
  - OpenMPTCProuter VPS version: 0.1028.5.4.207-mptcp
  - OpenMPTCProuter VPS provider: ARUBA NETWORK
  - OpenMPTCProuter platform: RPI4 model B

**Comment by Ysurac** [comment]
The red light flashing on a RPI indicate an undervoltage problem.
You have this only with one SIM in any dongle ? In this case you should really try a USB powered hub or another RPI power supply.
What is the SIM provider ?

**Comment by filmmaker75** [comment]
The red LED that flashes is that of the HUAWEI E3372-325. The RPI works properly, I use its original power supply. Again, if I connect two dongles it works perfectly. The problem is not the dongle but the SIM.
SIM 1 - Iliad
SIM 2 - HO.mobile
SIM 3 - Very mobile
Even trying them one at a time, the first two work while the third (very) doesn't. If I try it on my Windows PC it works correctly.

**Comment by R-sajadi** [comment]
LED on RPI or HUAWEI 3372-325 dongles?

**Comment by filmmaker75** [comment]
> LED on RPI or HUAWEI 3372-325 dongles?

The LED on the HUAWEI dongle flashes from red to green then red then green endlessly.
![huawei-e3372](https://github.com/Ysurac/openmptcprouter/assets/82170801/9cff4860-5f2c-4e8c-9200-b6d151a7fbe6)

**Comment by Ysurac** [comment]
ok, so not a power problem.
This dongle give directly some eth* interfaces or you need to use ModemManager ?

**Comment by filmmaker75** [comment]
> ok, so not a power problem.
> This dongle give directly some eth* interfaces or you need to use ModemManager ?

With working SIMs they are seen as USB0 and USB1 on openmptcprouter. However, if I insert the VERY SIM, no interface is created.

**Comment by Ysurac** [comment]
What do you have in Status->System log when you plug the dongle ?

**Comment by filmmaker75** [comment]
> What do you have in Status->System log when you plug the dongle ?

[system_log_very.txt](https://github.com/Ysurac/openmptcprouter/files/14199594/system_log_very.txt)

This is the log with the VERY SIM connected alone.

**Comment by Ysurac** [comment]
I see nothing here. Maybe in Status->kernel log ? Can you plug/unplug the dongle when running ?

**Comment by filmmaker75** [comment]
> I see nothing here. Maybe in Status->kernel log ? Can you plug/unplug the dongle when running ?

This is kernel log. I tried inserting and removing several times, even changing the USB port:

[kernel_log_very.txt](https://github.com/Ysurac/openmptcprouter/files/14199726/kernel_log_very.txt)

I also attach the kernel log of the two working sims if it can be useful to you. Thanks for your patience.

[kernel_log_iliad_ho.txt](https://github.com/Ysurac/openmptcprouter/files/14199730/kernel_log_iliad_ho.txt)

**Comment by Ysurac** [comment]
Ok, the dongle is not in the good mode (no idea why only on this SIM...).
Can you try, under the working PC, to upgrade the dongle by going to it's web interface (or maybe change mode from here, no idea if it's posssible) ?

**Comment by filmmaker75** [comment]
> Can you try, under the working PC, to upgrade the dongle by going to it's web interface (or maybe change mode from here, no idea if it's posssible) ?

From its web interface I don't have many options. If I send you the screenshots can you tell me what to change?

![Screenshot 2024-02-07 191447](https://github.com/Ysurac/openmptcprouter/assets/82170801/26c4deb2-bf0b-481d-8363-233060dcc6ab)
![Screenshot 2024-02-07 191538](https://github.com/Ysurac/openmptcprouter/assets/82170801/c08e7b2e-1173-4586-97df-6dbce19ed485)
![Screenshot 2024-02-07 191553](https://github.com/Ysurac/openmptcprouter/assets/82170801/edc270a7-b772-43b3-b16e-de5cf56611ba)
![Screenshot 2024-02-07 191605](https://github.com/Ysurac/openmptcprouter/assets/82170801/19bd3c30-2371-4976-ba9e-535069da8c60)
![Screenshot 2024-02-07 191615](https://github.com/Ysurac/openmptcprouter/assets/82170801/a94bcc82-2f98-461e-8121-b823e2cdd22c)
![Screenshot 2024-02-07 191627](https://github.com/Ysurac/openmptcprouter/assets/82170801/39a8be7f-9118-448c-8a21-92612af2e3e7)
![Screenshot 2024-02-07 191639](https://github.com/Ysurac/openmptcprouter/assets/82170801/0f410e53-e62c-4aa6-838d-d6ea4c3f8aca)
![Screenshot 2024-02-07 191647](https://github.com/Ysurac/openmptcprouter/assets/82170801/8fe4dd67-d925-43dc-8d9d-9df1a6bff901)
![Screenshot 2024-02-07 191654](https://github.com/Ysurac/openmptcprouter/assets/82170801/46b04cd2-7983-498e-9ea2-2b631606227b)
![Screenshot 2024-02-07 191700](https://github.com/Ysurac/openmptcprouter/assets/82170801/2d04ca5a-b3c1-498f-b94e-2251224c84f0)
![Screenshot 2024-02-07 191721](https://github.com/Ysurac/openmptcprouter/assets/82170801/a8422dbe-ff6d-4577-80c4-7513f71f4ca4)
![Screenshot 2024-02-07 191728](https://github.com/Ysurac/openmptcprouter/assets/82170801/190cc35a-6ea7-45a1-a16f-7b586d6744d3)
![Screenshot 2024-02-07 191752](https://github.com/Ysurac/openmptcprouter/assets/82170801/9a1f0039-7539-4a94-a1f4-1380e509039c)
![Screenshot 2024-02-07 191801](https://github.com/Ysurac/openmptcprouter/assets/82170801/fd44bc75-828f-4e54-b7d3-a65f824e0d39)

**Comment by Ysurac** [comment]
Can you try to put this file 
[usb-mode.json](https://github.com/Ysurac/openmptcprouter/files/14200082/usb-mode.json) in ```/etc/``` on the router and reboot it ?

**Comment by filmmaker75** [comment]
> Can you try to put this file [usb-mode.json](https://github.com/Ysurac/openmptcprouter/files/14200082/usb-mode.json) in `/etc/` on the router and reboot it ?

Of course I can try but you have to explain to me how to do it. I don't know how to do it.

**Comment by Ysurac** [comment]
Via SSH, using scp command line or WinSCP under windows

**Comment by filmmaker75** [comment]
Done. It asked me to overwrite the old file and I confirmed. Restarted everything but same situation.

**Comment by Ysurac** [comment]
I have no more idea...
Really strange that you have issue only with a SIM card...
On this type of dongle, that are using HiLink mode, all is done on modem side. For OpenMPTCProuter it's like an ethernet device.
The json I gave you should force the dongle to switch to modem mode instead of storage mode, but this doesn't work.
It's like if the SIM doesn't allow the dongle to be in modem mode... Don't know if it's possible if the key is not customized by provider...

**Comment by filmmaker75** [comment]
> I have no more idea...
> Really strange that you have issue only with a SIM card...
> On this type of dongle, that are using HiLink mode, all is done on modem side. For OpenMPTCProuter it's like an ethernet device.
> The json I gave you should force the dongle to switch to modem mode instead of storage mode, but this doesn't work.
> It's like if the SIM doesn't allow the dongle to be in modem mode... Don't know if it's possible if the key is not customized by provider...

The key is not personalized. Even by changing the key the situation does not change. By inserting one of the working SIMs into this one, it works too. So the problem is definitely on the SIM. I will do some tests with some Linux distributions to verify how they work but my knowledge of Linux is zero. Thanks anyway for everything.

**Comment by filmmaker75** [comment]
I'll update the post with a test I did out of curiosity. I prepared a pendrive with a live debian distro. I started it on my PC and connected the HUAWEI dongle with the Very SIM. And it doesn't work, it has the same behavior as on the RPI4 with OMR. So I assume that there is something blocking it on Linux's part. Obviously I also tried inserting the other two SIMs and, as on OMR, they work perfectly. Evidently there is some different command to give or some sort of incompatibility between Linux and the Very SIM.
